[System] Add offset feature to Grid (#33415)
mui/material-ui: docs/data/system/components/grid/grid.md

{{"demo": "ColumnsGrid.js", "bg": true}}

## Offset

> Move the item to the right by using offset props which can be a number or `"auto"`.

Reviewer: ```suggestion
Move the item to the right by using offset props which can be:
```

> - a number, for example, `mdOffset={2}` moves the item to the right by 2 columns starts from `md` breakpoint and up.
> - The `"auto"` value moves the item to the right edge of the grid container.

Reviewer: ```suggestion
- number, for example, `mdOffset={2}` - when used the item is moved to the right by 2 columns starts from `md` breakpoint and up.
- `"auto"` - when used, the item is moved to the right edge of the grid container.
```
